scripts/run-nix-examples.sh

```diff
@@ -1,20 +1,28 @@
 #!/bin/bash
 
+SourceDir="$(dirname $0)/.."
 BaseDir="$(pwd)"
+
 TestAppDir="$(pwd)/tiledb/examples/c_api"
 TestAppDataDir="$(pwd)/tiledb/examples/c_api/test_app_data"
-for exampleexe in $(ls ${TestAppDir}/*_c) ;
+for example in $(ls ${SourceDir}/examples/c_api/*.c) ;
 do
   cd ${TestAppDir}
   rm -rf ${TestAppDataDir}
   mkdir ${TestAppDataDir}
   cd ${TestAppDataDir}
-  echo $exampleexe
-  $exampleexe;
+  exampleexe=${example%.c}_c
+  exampleexe=${exampleexe##*/}
+  cmake --build ${BaseDir}/tiledb --target ${exampleexe}
+  echo $TestAppDir/$exampleexe
+  $TestAppDir/$exampleexe;
+  # Remove the executable after running it to prevent disk
+  # space exhaustion when statically linking to tiledb.
+  rm $TestAppDir/$exampleexe
   status=$?
```

Comment:
Shouldn't the status be checked before the added `rm` command?

Reply:
Oops, reordered. Thanks!